### ISSUE TITLE
cryptonight algorithm performs poorly compared to xmrig-nvidia

### ISSUE THREAD

**Issue by gordan-bobic** [nicehash/excavator#118]
Excavator's cryptonight performance is relatively poor poorly compared to xmrig-nvidia. Performance difference is less than 10% on a 1070Ti, but nearer 30% on a 980Ti.

1070Ti:
excavator: 650H/s
xmrig-nvidia: 700H/s

980Ti:
excavator: 490H/s
xmrig-nvidia: 800H/s

**Comment by dropky** [comment]
@gordan-bobic thanks for the feedback! We're aware of the performance gap and we're constantly working hard to improve excavator. Unfortunately, the code  for cryptonight hasn't been fully optimized yet. Currently, our main focus is on neoscrypt, nist5 and equihash, but we'll definitely return to cryptonight very soon.

**Comment by gordan-bobic** [comment]
If you are planning to address the issue, shouldn't this ticket remain open until you do?

**Comment by dropky** [comment]
We already have this in our backlog and since it's not a bug I don't see any benefits in remaining it open. It's easier for us to track new issues and keep the track of the ones that still need our attention if there is less opened issues. I hope you understand.

**Comment by dropky** [comment]
But yeah.. I can reopen it :) Not a problem.

**Comment by gordan-bobic** [comment]
I totally understand if you use a separate issue tracking system. It's just that then I as a user can't tell when it's  been fixed so I can look out for a new build inbound. :-)

**Comment by Sephiroth4269** [comment]
My issue is, I cannot find the file that allows me to change or update my CUDA paramters. I have a Titan V that is mining on all of the algoritms, but isnt submitting any shares. Can you please advise?

**Comment by dropky** [comment]
@Sephiroth4269  Difficulty for cryptonight is quite high, it might take a while before the miner finds any shares (especially if you're mining with one GPU). What hashrate do you get?

**Comment by gordan-bobic** [comment]
@Sephiroth4269 the short answer to this one is - don't mine cryptonight on nicehash unless you have an AMD Vega. Difficulty set by the stratum is so high that by the time you submit any shares the profitability peak you were hoping to surf will be long gone, and if you don't submit any hashes you won't get anything for it.

**Comment by Sephiroth4269** [comment]
When I launched 2.0.5 I was getting 1.5 MH/s when I updated to the latest version....it dropped massively to 1.5kH. I wanted to play around with the files that determine Parameters, as well as figure out where exactly is the "update" to the CUDA cores. Because I wasn't able to "share" any of my work even after 2 hours of mining not a single "net | Share ___ Accepted" line showed up. I was assuming that had more to do with CUDA communication rather than "difficulty"?

**Comment by gordan-bobic** [comment]
@Sephiroth4269 
We are clearly not talking about the same algorithm here, then. This thread is specifically about cryptonight. Fastest CUDA capable card tuned to the eyeballs will make about 850H/s on cryptnight (not KH/s, not MH/s, H/s). Fastest mining devices for cryptonight are AMD Vega, and they deliver at most 2KH/s. Excavator doesn't run on AMD cards.

Perhaps open a separate issue to track the exact algorithm problem you are having? It sounds like you may be talking about a neoscrypt performance regression in 1.4.4. I did open an issue about that, but it has been closed.

**Comment by Sephiroth4269** [comment]
![proof](https://user-images.githubusercontent.com/36573672/36345676-8f8745f8-13fd-11e8-8af7-3cfc24bd628d.png)

This is what I am talking about. That is the algorithm and the results I am getting and I was trying to make sure that it wasnt due to a lack of CUDA compatibility. It ran for 3 hours without a single share at that rate. I wanted to know how to address it and what determines "share rates", whether that's CUDA cores, parameters, Clock Speed, or whatever.

**Comment by papacheal** [comment]
Not sure if it's a version issue of Excavator or not, but your build date shows 2017-11-24 and the Excavator release that official indicated Titan V support was 1.4.3a, released at the beginning of Feb/2018. You may be using the version baked into the NiceHash Miner tool, but thought I'd mention it.

**Comment by gordan-bobic** [comment]
There has to be an error somewhere. 1.2KH/s from a Titan V is plausible. 1.2MH/s is not.

**Comment by dropky** [comment]
That's right, @gordan-bobic thanks for pointing it out. As mentioned before v1.3.8 doesn't support 7.0 architecture nor CUDA 9. Official support for Titan V was added with version 1.4.3.
